### ISSUE TITLE
Open strings in the user preferred Translation editor mode on the search page and browse page

### ISSUE THREAD

**Issue by Geeyun-JY3** [WeblateOrg/weblate#7470]
### Describe the problem

For now, whatever Translation editor mode the user chose, the link of strings in the search page and browse page always open the full editor.

### Describe the solution you'd like

Open strings in the user preferred Translation editor mode on the search page and browse page

### Additional context

Following #1029

**Comment by nijel** [comment]
Makes sense, using template tag is really error-prone. When changing that, please keep in mind that in some cases the links to /translate are intentional.

**Comment by ruchi-mish90** [comment]
Taking this one - using the `translation_editor` redirect @wokepadia suggested. Adding an `is_authenticated` guard before the `.profile` check since anon users don't have one. Will keep the intentional `/translate` links as-is. Draft PR coming once it works.
